vendor/github.com/containernetworking/cni/libcni/api.go

```diff
@@ -15,14 +15,27 @@
 package libcni
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 	"strings"
 
 	"github.com/containernetworking/cni/pkg/invoke"
 	"github.com/containernetworking/cni/pkg/types"
 	"github.com/containernetworking/cni/pkg/version"
 )
 
+var (
+	CacheDir = "/var/lib/cni"
```

Comment:
does this require this directory be writeable? what happens if the kubelet runs in an environment where this is not writeable?

Reply:
hmm, looks like AddNetwork will return nil instead of result even cni plugin succeeded, with following message:
("failed to set network %q cached result: %v", net.Network.Name, err)
from my point of view it's better to return the result, even access to cache was failed or check cache dir at libcni initialization time.
Looks like after removing nsenter, cni cache became critical for kubelet, especially after kubelet restart. Maybe set CacheDir as a kubelet command line option.

Reply:
it was bad idea to suggest return result in case when cache failed, in this case pod data could be in inconsistent state, we will lost ip address after kubelet restart.

Reply:
@liggitt yes, it requires the directory to be writable. Note that a default install of kubelet using the kubenet plugin also requires a writable /var/lib/cni/networks for host-local already. We could certainly change the default cache dir to be /var/lib/cni/<something> if that's more helpful.

Reply:
I'll defer to sig-node and sig-network to figure out what a guaranteed-writeable dir is

Reply:
a related question is what handles cleanup of stale data in this cache dir? are there changes required to things like `kubeadm reset` to clear this out when resetting a node, or does stale data here get pruned automatically by CNI?